daft/expressions/expressions.py

```diff
@@ -660,3 +665,8 @@ def get_expression_by_name(self, name: str) -> Expression:
     def resolve_schema(self, schema: Schema) -> Schema:
         fields = [e._to_field(schema) for e in self]
         return Schema._from_field_name_and_types([(f.name, f.dtype) for f in fields])
+
+
+class ExpressionImageNamespace(ExpressionNamespace):
+    def decode(self, mode: ImageMode, height: int, width: int) -> Expression:
```

Comment:
Decoding a binary array into images requires us to know the mode, height, and width (so we know the dtype and shape of of the data); we could default to RGB and u8, but that could silently fail. I thought that `(mode, height, width)` as required args would be better than `DataType.image()`, which currently has these three fields as optional.